lte/gateway/c/oai/tasks/amf/amf_app_pdu_resource_setup_req_rel.cpp

```diff
@@ -171,15 +271,43 @@ int pdu_session_resource_release_request(
       NORMAL_RELEASE;
   amf_pdu_ses_rel_transfer_req.cause.cause_group.cause_group_type = NAS_GROUP;
 
-  OAILOG_DEBUG(
-      LOG_AMF_APP, "filling pdu_session_resource_release_command_transfer\n");
+  // Convert amf_pdu_ses_setup_transfer_req to bstring and assign to message_p
+
+  OAILOG_INFO(
+      LOG_AMF_APP,
+      //"Converting pdu_session_resource_release_command_transfer to
+      // bstring\n");
+      "filling pdu_session_resource_release_command_transfer\n");
+  // ngap_pdu_ses_release_req->pduSessionResourceToRelReqList.item[0]
+  //    .PDU_Session_Resource_TO_Release_Command_Transfer = blk2bstr(
+  //    &amf_pdu_ses_rel_transfer_req,
+  //    sizeof(
+  //        pdu_session_resource_release_command_transfer));  //
+  //        pdu_res_set_change
   ngap_pdu_ses_release_req->pduSessionResourceToRelReqList.item[0]
       .PDU_Session_Resource_TO_Release_Command_Transfer =
       amf_pdu_ses_rel_transfer_req;
-
   // Send message to NGAP task
+  if (bytes > 0) {
+    OAILOG_INFO(LOG_AMF_APP, "NAS encode success for PDU Release Command\n");
+    //    buffer->slen = bytes;
+    buffer->slen = bytes;
+    for (int i = 0; i < blength(buffer); i++) {
+      OAILOG_INFO(LOG_AMF_APP, "buffer[%d]: %x", i, buffer->data[i]);
+    }
+    //    bytes + 3;  // TODO fix length in all other DownlinkNASTransport
+    // amf_app_def_as.amf_app_handle_nas_dl_req(ue_id, buffer, rc);
+    //    amf_app_handle_nas_dl_req(ue_id, buffer, rc);
+    ngap_pdu_ses_release_req->nas_msg = bstrcpy(buffer);
+    bdestroy(buffer);
+  } else {
+    bdestroy(buffer);
```

Comment:
if there will be no return in this clause `bdestroy` on lines 302 and 304 can be moved out to if/else clause. However, if NAS encode has failed, do we still want to send msg and return RETURNok?

Reply:
if returning in else clause, we need to free the allocated message.